### ISSUE TITLE
adding a max_count for a child page?

### ISSUE THREAD

**Issue by JaredOzzy** [wagtail/wagtail#5065]
I would like the ability to have the max_count on a specific child.

for example i have a Page with child pages, each of the child pages can have 1 detail page of their own and many other types of pages.

here is an example of what i mean. Apologies if its difficult to understand what im trying to ask.
```
class MainPage(PagePlus):
    text = RichTextField()

    content_panels = Page.content_panels + [
        FieldPanel('text'),
    ]

    parent_page_types = ['HomePage']


class DetailPage(PagePlus):
    pass
    parent_page_types = ['MainPage']
    max_count_per_child = 1 <<<<<<<<<<<<<<<<<<<<<

class AnotherPage(Page):
    pass
    parent_page_types = ['MainPage']
```
now in the CMS it would work similarly to the newly added max_count, but only allow 1 detail page per child created. currently if i added the max_count it would allow me to make 1 detail page accross all the child pages, but i would like 1 detail page per child page.

Just a possible addition that could help in some specific cases.

**Comment by zerolab** [comment]
FWIF, you can achieve this as seen in the tests - https://github.com/wagtail/wagtail/blob/master/wagtail/tests/testapp/models.py#L865-L870 and a bit of logic per page type.

**Comment by gasman** [comment]
Hi @JaredOzzy - thanks for the suggestion. As @zerolab mentions, this can be achieved with a custom [`can_create_at`](https://docs.wagtail.io/en/stable/reference/pages/model_reference.html#wagtail.core.models.Page.can_create_at) method, but I expect that this is a common enough setup that it would be worth adding a standard 'shortcut' for this like `max_count`. (I'm not sure about the name, though - `max_count_per_parent` would probably be more accurate...)

**Comment by JaredOzzy** [comment]
@zerolab @gasman thanks! will give it a shot.

**Comment by leewesleyv** [comment]
I hope I didn't take the work from you, but I really wanted to try this out!

**Comment by JaredOzzy** [comment]
> I hope I didn't take the work from you, but I really wanted to try this out!

Not at all! glad you got it working, will be using your work once its live! ;)

**Comment by gasman** [comment]
Implemented in #5151.
